### ISSUE TITLE
APNs not registering (cordova-plugin-firebasex vs cordova-plugin-firebasex-test)

### ISSUE THREAD

**Issue by lukasza67** [dpa99c/cordova-plugin-firebasex#604]
Hello and thanks again for a fantastic job! I have spend over 10 hrs of reviewing all the config files and version dependencies but cannot really find the error which led my conclusions that there might be a recent bug/decency injected in the plugin.

I'am running latest everything to date (2021-03-24).

The final issue in the stack is that ANPs is not getting registered and onMessageReceived not being fired (iOS only). 

(project 1)
When forking the cordova-plugin-firebasex the ordinary way into a clean project the issue is a fact. 

(project 2)
When cloning the cordova-plugin-firebasex-test, and using my original config.xml, and using my own GoogleServises-Info.plist. the logics in www of the test project works, and when using my own app code it even works.

(project 3)
I then did the reverse and copied (project 2 --> project 1) package.json (run npm install in later the end), and made config.xml identical. I purged plugins and platforms folders and added everything freshly over again. I even "ls" the plugin list in project 2 and added all the missing plugins into project 1. This still did not work. 

I will repeat that identical Firebase/GoogleServises-Info.plist has been tested in all 3 projects with 1/3 success rate. So I ruled out Firebase/Keys config might be wrong. 

I am sorry I know that this is a pretty much black box tested situation, But I had been playing with two repositories here and  cannot come up with any other solution that to use the test-project for production.

A the moment I simply cannot manage to do an out-of-the-box successful run from the clean cordova-plugin-firebasex fork (which I think is the right way to go instead of using the test repo).

It could be good if someone at your end could do a clean fork of the plugin and run the FirebasePlugin.functions() without any test repo involvement. 

Many thanks again for the great work that you have done!

**Comment by dpa99c** [comment]
I regression tested all plugin functionality on both platforms with the [example project](https://github.com/dpa99c/cordova-plugin-firebasex-test) as part of the v12 major release including push notifications on iOS and they worked as expected.
Therefore I'm closing this on the basis that the issue is specific to your configuration.
However if you can provide evidence that this is a more generic issue with the plugin affecting other users, I'll be happy to reopen and investigate.

**Comment by lukasza67** [comment]
Ok Thanks for the help. I figured this out now and what I did. Your test project code 'www' in not solely compatible running on iOS clean cordova project but it is on Android and the reason for this is that the custom message plugin was missing. What is weird is that one custom fcm clause is breaking the whole iOS test but Android just surpass it and continues executing like nothing happened. Regards.
